File: src/main/java/br/com/alura/forum/model/Role.java
package br.com.alura.forum.model;

import java.util.Objects;

import javax.persistence.Entity;
import javax.persistence.Id;

import org.springframework.security.core.GrantedAuthority;

@Entity
public class Role implements GrantedAuthority {

	public static final Role ROLE_ADMIN = new Role("ROLE_ADMIN");
	public static final Role ROLE_ALUNO = new Role("ROLE_ALUNO");

	private static final long serialVersionUID = -3430174216330293377L;

	@Id
	private String authority;

	/**
	 * @deprecated
	 */
	public Role () {}

	public Role(String authority) {
		this.authority = authority;
	}

	@Override
	public String getAuthority() {
		return this.authority;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		Role role = (Role) o;
		return Objects.equals(authority, role.authority);
	}
}

File: src/main/kotlin/br/com/alura/forum/model/Role.kt
package br.com.alura.forum.model

import org.springframework.security.core.GrantedAuthority
import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class Role(@Id private val authority: String) : GrantedAuthority {

    override fun getAuthority() = authority

    companion object {
        const val ALUNO = "ROLE_ALUNO"
        const val ADMIN = "ROLE_ADMIN"
    }
}